src/gt4sd/frameworks/enzeptional/processing.py

```diff
@@ -242,16 +378,76 @@ def reconstruct_sequence_with_mutation_range(
         reconstructed sequence.
     """
     # create the mutated sequence, considering sorted intervals
-    mutated_range_offset = 0  # offset with respect to the mutated_sequence_range
-    mutated_sequence_offset = 0  # offset with respect to the full mutated sequence.
-    mutated_sequence = ""
+    mutated_sequence = list(sequence)
+    # split fragments by intervals
     for start, end in intervals:
-        mutated_sequence += sequence[mutated_sequence_offset:start]
-        chunk_length = end - start + 1
-        mutated_sequence += mutated_sequence_range[
-            mutated_range_offset : mutated_range_offset + chunk_length
-        ]
-        mutated_range_offset += chunk_length
-        mutated_sequence_offset = end + 1
-    mutated_sequence += sequence[end + 1 :]
-    return mutated_sequence
+        size_fragment = end - start
+        mutated_sequence[start:end] = list(mutated_sequence_range[:size_fragment])
+        mutated_sequence_range = mutated_sequence_range[size_fragment:]
+    return "".join(mutated_sequence)
+
+
+def selection(scores: List[Dict[str, Any]], k: int = 20) -> List[Any]:
+    """Select the top k mutated sequences based on the score
+
+    Args:
+        scores: dictionary containing sequences and scores.
+        K: number of top sequences to return
+
+    return:
+        retuen the top K sequences
+
+    """
+
+    res = list(sorted(scores, key=lambda d: d["score"], reverse=True))[:k]
+
+    return res
+
+
+def single_point_crossover(sequence_1: str, sequence_2: str) -> Tuple[str, str]:
```

Comment:
follow the style of other gt4sd functions, do not use _NUMBER but `a_sequence`, `another_sequence`